frontend/src/components/UserProfile/UpstreamProviderList.tsx

```diff
@@ -0,0 +1,127 @@
+// Copyright 2024 New Vector Ltd.
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+// Please see LICENSE in the repository root for full details.
+
+import { H5 } from "@vector-im/compound-web";
+import { useCallback, useTransition } from "react";
+import { useTranslation } from "react-i18next";
+import { useQuery } from "urql";
```

Comment:
fyi, I'm in the process of moving off urql to @tanstack/react-query for querying data. This shouldn't make a massive difference, you'll have basically three things to change:

 - useQuery calls have a slightly different pattern (see https://github.com/element-hq/matrix-authentication-service/pull/3504/files#diff-824d80f55dbfb304574729bfc83deb51ec11af2ed3e3467cbd132c10ffaafffd )
 - useMutation calls are also slightly different (see https://github.com/element-hq/matrix-authentication-service/pull/3504/files#diff-79ea95d633c486de54ab8a248faa3ba49b217bd739d5a22644b982743dd0544e )
 - there are now 'render' snapshot tests, so you'll need to add the additional fragments to the mocked GraphQL response here: https://github.com/element-hq/matrix-authentication-service/pull/3504/files#diff-74973935dbe4236af3173d6b7989acbb33a1318c44a5175ac9d573b825b9afa8R99